### DISCUSSION TITLE
计划、标签和程序之间的计时逻辑是否重构下？

### DISCUSSION THREAD

**Discussion by Ricardolee-Hu** [shion-app/shion#55] [Q&A] [answered]
我有点不太明白计划、标签作为手动输入的信息，需要手动开启计时的同时，其他的程序也会同时自动被计时，这个背后的考虑是什么，这样会出现重复计时吧？从我自己的感受来说，我去定义一个计划，它可能是需要用一个或者多个程序去完成的，比如我设定了阅读计划，标签或者其他什么tag可以是对应的阅读软件，那当我手动开启这个计划之后，那就应该在我使用阅读软件的时长或者我手动开启的时长之中只记录一个的时长，建议是否重构下，我设想有几个路径：
1.在创建计划的时候，标签就是直接对应监听的应用程序，从列表中筛选1个或多个程序，定义为计划的载体，计划的计时可以手动开启，开启之后，对应列表中的程序使用时间聚合为计划的投入时间；
2.在1的基础上，如果能够加入自动识别和分类功能就更好了，比如在计划的设定时间范围内，无论计划的计时手动开启与否，只要打开对应的程序，则自动计入计划用时，如果能够更进一步识别标签对应网页、对应文档就更好了，这样就可以定义浏览器和阅读软件在打开不同的网页、不同的文档的时候，处理的是不同的任务了；
3.或者计划与程序之间仍然没有对应关系，那么将这两类时长拆分为两个维度，而不是柱状图的一个柱子里体现，每天使用各个程序的时间和在各个计划里投入的时间分开显示；
以上建议，供参考。

**Comment by hanaTsuk1** [comment, marked as answer]
> 这样会出现重复计时吧

你自己有一个心理预期就是计时**不能重复**，但实际情况却不是这样
比如，我在听歌的时候打游戏，两个行为就是重叠了，它的表现形式可能看上去怪异，但你把它放在时间线的维度就很正常
同理，手动计时和自动计时出现重叠也一样，只不过这时你想让手动计时表现为描述自动计时的行为，而不是一个单独的行为

> 那么将这两类时长拆分为两个维度，而不是柱状图的一个柱子里体现

即使拆分也不会解决一小时内总和在这个范围内


总的来说你有一套思维模式，但使用软件的时候你需要跟着设计者的考虑走
感谢你的提议，不知道这样的解释你觉得怎样🧐

**Reply by Ricardolee-Hu** [reply]
好的，谢谢你的反馈，你的孩子我们可以喜欢，但最终由你塑造，哈哈
